client-react/src/pages/app/functions/app-keys/HostKeys.tsx

```diff
@@ -196,20 +219,40 @@ const HostKeys: React.FC<HostKeysProps> = props => {
         return <></>;
       }
       return (
-        <TooltipHost
-          content={t('delete')}
-          id={`app-keys-host-keys-delete-tooltip-${index}`}
-          calloutProps={{ gapSpace: 0 }}
-          closeDelay={500}>
-          <IconButton
-            className={defaultCellStyle}
-            disabled={readOnlyPermission}
-            id={`app-settings-application-settings-delete-${index}`}
-            iconProps={{ iconName: 'Delete' }}
-            ariaLabel={t('delete')}
-            onClick={() => deleteHostKey(itemKey)}
-          />
-        </TooltipHost>
+        <div>
+          <TooltipHost
+            content={t('delete')}
+            id={`app-keys-host-keys-delete-tooltip-${index}`}
+            calloutProps={{ gapSpace: 0 }}
+            closeDelay={500}>
+            <IconButton
+              className={defaultCellStyle}
+              disabled={readOnlyPermission}
+              id={`app-settings-application-settings-delete-${index}`}
+              iconProps={{ iconName: 'Delete' }}
+              ariaLabel={t('delete')}
+              onClick={() => deleteHostKey()}
```

Comment:
Bind directly to `deleteHostKey`.

```suggestion
              onClick={deleteHostKey}
```